packages/sanity/src/core/bundles/components/panes/BundleActions.tsx

```diff
@@ -1,51 +1,60 @@
 import {AddIcon, CheckmarkIcon} from '@sanity/icons'
 import {useToast} from '@sanity/ui'
-import {type ReactNode, useCallback, useEffect, useState} from 'react'
+import {type ReactNode, useCallback, useState} from 'react'
 import {filter, firstValueFrom} from 'rxjs'
 import {useDocumentOperation, useDocumentStore} from 'sanity'
 
 import {Button} from '../../../../ui-components'
 import {type BundleDocument} from '../../../store/bundles/types'
-import {versionDocumentExists} from '../../util/util'
+import {getBundleSlug} from '../../util/util'
 
 interface BundleActionsProps {
   currentGlobalBundle: BundleDocument
+  /**
+   * Base document id, the published one
+   */
   documentId: string
+  /**
+   * The id of the document that is displayed in the form, might be a draft or a version
+   */
+  formDocumentId: string
```

Comment:
Instead of waiting for `documentVersions` which fetches the documents, we can check the form document id, if it has the slug of the current global bundle, then the button should indicate that this document already exists.

Reply:
could it be named `versionId`?